src/components/bot-challenge-protection/index.jsx

```diff
@@ -21,15 +21,21 @@ const BotChallengeProtection = ({
 }) => {
 	const { isInitialized } = useIdentity();
 	const { recaptchaVersion, siteKey, isRecaptchaEnabled } = useRecaptcha(challengeIn);
+	const [ currentRecapctha, setCurrentRecaptcha ] = useState();
 
 	const recaptchaRef = React.createRef();
 
+	useEffect(()=>{
+		const {current} = recaptchaRef;
+		setCurrentRecaptcha(current);
+
+	},[recaptchaRef]);
+
```

Comment:
I'm not understanding why this file is being changed at all. Also this is the same as including `recaptchaRef` in the other dependency array, plus causing an additional re-render. So it's pretty bad for performance.

Reply:
Hi @vgalatro I was seeing a weird behavior while I was using this component in our checkout block, and it's because recaptchaRef is changing every time the user completes the challenge, it'is causing the reset again. In order to avoid the linting you mentioned in a previous comment, the only way that I found to handle this, was by destructuring the response and just check current.

Reply:
I just realized it's not using the hook for the reference. Have you tried `const recaptchaRef = useRef();` instead of `const recaptchaRef = React.createRef();`?